src/ApiCore/TransportFactory.php

```diff
@@ -55,20 +56,16 @@ class TransportFactory
         'transport'         => null,
         'authCacheOptions'  => null,
         'httpHandler'       => null,
-        'authHttpHandler'   => null,
-        'transport'         => null
+        'authHttpHandler'   => null
     ];
 
     /**
      * Builds a transport given an array of arguments.
      *
-     * @param array $args {
+     * @param string $transport The type of transport to build.
```

Comment:
Since it is optional, what is the argument for making `transport` its own parameter?

Reply:
Hmm, you are right. I was treating it as required, but that is not correct.